modules/jaw/jaw.psm1

```diff
@@ -38,38 +38,34 @@ function Expand-Template {
   Searches for dockerfiles and saves information about them for later use.
 #>
 function Set-k8sConfig {
-    [CmdletBinding(SupportsShouldProcess = $true)]
-    param(
-        [string] $AppPath,
-        [string] $OutPath
-    )
-    # parse each Dockerfile in to a K8S JSON
-    # we only check for top level dockerfiles right now.
-    Get-ChildItem -Path $AppPath -Filter "*dockerfile*" -Recurse `
-    | % {
-        # parse each Dockerfile directive into @{$command => $args}
-        $commands = @{ }
-        Get-Content $_.FullName `
-        | ? { $_ -notmatch "^\s*$" } `
-        | % {
-            # First word is cmd, all else is args
-            $words = $_ -split " "
-            $cmd = $words | Select -First 1
-            $args = $words | Select -Skip 1
+  [CmdletBinding(SupportsShouldProcess = $true)]
+  param(
+    [string] $AppPath,
+    [string] $OutPath
+  )
+  # parse each Dockerfile in to a K8S JSON
+  # we only check for top level dockerfiles right now.
+  Get-ChildItem -Path $AppPath -Filter "*dockerfile*" -Recurse `
+  | % {
+    # Full File Name
+    $Name = $_.FullName
+    # Parent Directory
+    $Path = $_.Directory
+    # This resolves to the folder directly beneath /app
+    $ImageName = ($Path -split "/")[-1]
+    # Query inactive container for config
+    $Config = docker inspect mics233.azurecr.io/$ImageName --format="{{json .Config}}"
+    # Extract port numbers out of config
+    $Ports = ($Config | ConvertFrom-Json -AsHashtable).ExposedPorts.Keys | % { ($_ -split "/")[0] }
```

Comment:
This line has a lot of idiomatic Powershell in it.  

`($Config | ConvertFrom-Json -AsHashtable)` takes the config variable (a json string we got from docker inspect) and passes it to the ConvertFrom-Json cmdlet to deserialize it.  We specifically want it deserialized as a hashtable instead of a generic object.  

`.ExposedPorts` is a member on the resulting object.  It is a hashtable that has Port numbers mapped to null strings (not sure why, its just how docker inspect gives it to us).  `.Keys` selects all the key values which will look something like `80/tcp`.  

We pipe these keys on to `% { }`, which is a Powershell foreach loop.  The `$_` variable represents the current key in the array.  So for each key in keys, split the key at the `/` character.  This gives us `80, tcp`.  We only care about the port number, so we take [0] and discard the rest.

$Ports is an array of port numbers as strings.